src/main/kotlin/org/elm/ide/test/run/ElmTestLocator.kt

```diff
@@ -0,0 +1,80 @@
+package org.elm.ide.test.run
+
+import com.intellij.execution.Location
+import com.intellij.execution.PsiLocation
+import com.intellij.execution.testframework.sm.FileUrlProvider
+import com.intellij.execution.testframework.sm.TestsLocationProviderUtil
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.io.FileUtil
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.PsiDocumentManager
+import com.intellij.psi.PsiManager
+import com.intellij.psi.search.GlobalSearchScope
+import org.elm.ide.test.core.ElmPluginHelper
+import org.elm.ide.test.core.LabelUtils
+
+
+class ElmTestLocator private constructor() : FileUrlProvider() {
+
+    override fun getLocation(protocol: String, path: String, metainfo: String?, project: Project, scope: GlobalSearchScope): List<Location<*>> {
+        if (!protocol.startsWith(LabelUtils.ELM_TEST_PROTOCOL)) {
+            return super.getLocation(protocol, path, metainfo, project, scope)
+        }
+
+        if (protocol.startsWith(LabelUtils.ERROR_PROTOCOL)) {
+            val pair = LabelUtils.fromErrorLocationUrlPath(path)
+            val filePath = pair.first
+            val line = pair.second.first
+            val column = pair.second.second
+
+            val systemIndependentPath = FileUtil.toSystemIndependentName(filePath)
+            val virtualFiles = TestsLocationProviderUtil.findSuitableFilesFor(systemIndependentPath, project)
+            return if (virtualFiles.isEmpty()) {
+                emptyList()
+            } else virtualFiles
+                    .map { getErrorLocation(line, column, project, it) }
+                    .filter { it != null }
+                    .map { it!! }
```

Comment:
Seems that this could be simplified to
`return virtualFiles.mapNotNull { getErrorLocation(line, column, project, it) }`

Reply:
Thanks.